File: src/main/java/ru/capjack/degos/publish/DegosPublishExtension.java
package ru.capjack.degos.publish;

public interface DegosPublishExtension {
	String NAME = "degosPublish";
	
	boolean isAsPrivate();
	
	void setAsPrivate(boolean value);
	
	String getUsername();
	
	void setUsername(String username);
	
	String getPassword();
	
	void setPassword(String password);
}

File: src/main/kotlin/ru/capjack/degos/publish/DegosPublishExtension.kt
package ru.capjack.degos.publish

open class DegosPublishExtension {
	var private: Boolean = false
	var username: String? = null
	var password: String? = null
	var publication: String? = null
	var publicationSources: PublicationSource = PublicationSource.ALWAYS
	
	enum class PublicationSource {
		ALWAYS,
		RELEASE,
		NEVER
	}
}
